src/frontend/src/instance.rs

```diff
@@ -543,6 +570,20 @@ impl PromqlHandler for Instance {
     }
 }
 
+pub fn validate_catalog_and_schema(
```

Comment:
If we treat this as early form of our permission system, we can name this function as `check_permission` and accepts `&statement` and `query_ctx`

Reply:
And also `plugins`. Move `need_validate` into this function as well.